R/html_document_base.R

```diff
@@ -65,6 +66,23 @@ html_document_base <- function(theme = NULL,
     if (is_bs_theme(theme)) bslib::bs_global_set(old_theme)
   }
 
+  # additional sass/css
+  for (f in css) {
+    if (is_bs_theme(theme)) {
+      theme <- bslib::bs_add_rules(theme, sass::sass_file(f))
+      next
```

Comment:
Trying to understand how this will be different than the current behavior: 

`sass_file()` will take any CSS and SCSS file to be processed by **bslib** and no more passed to Pandoc is that right ? not just sass file ?

It won't be as simple as before, like changing header to red  

CSS in `style.css`
```css
h1 {
  color: red;
}
```
With standard theme it works ok
````markdown
---
output: 
  html_document:
    self_contained: false
    css: style.css
---

# A title that I want red

Some text
```` 
With bootstrap 4, I need to tweak more because CSS won't apply as before ? 
````markdown
---
output: 
  html_document:
    self_contained: false
    css: style.css
    theme:
      version: 4
---

# A title that I want red

Some text
````

Is that expected ?

Reply:
> With bootstrap 4, I need to tweak more because CSS won't apply as before ? Is that expected ?

I'm not sure what you mean. The difference is that, in the 2nd example, the CSS files are essentially appended to the Bootstrap stylesheet (instead of being processed by Pandoc), but you should still get the same end result. If you're worried about the implications for CSS precedence, we could process CSS files with Pandoc instead, but scss/sass files definitely want to be added via `bslib::bs_add_rules(theme, sass::sass_file(f))`

Reply:
I take the opportunity of this PR to test a little how all this works. 

> Your examples give the same end result

I don't get the same end result - h1 title is not red for me in the second example. Note the `self_contained: false`. If I set to TRUE then I get the red title. Not sure why this difference - CSS does not seem to be included correctly in not self contained mode ? Is this expected or a bug ?

> If you're worried about what this difference means for CSS precedence

I am noticing indeed that the precedence is different: 

This won't change the title in red
```yaml
output: 
  html_document:
    css: style.css
    theme:
      bootswatch: cerulean
```

This will 
```yaml
output: 
  html_document:
    css: style.css
    theme: cerulean
```

Is that because the bootstap rules are not overridden by a custom CSS as it was before ?

We have different behavior that I am trying to understand. Maybe it is expected but in this case it creates confusion some stuff that already works with Rmarkdown won't work the same with new bootstrap 4 support. This means documentation load for sure in this case. cc @apreshill

Reply:
Yikes, `sass::sass_file()` wasn't working the way I thought it did. Switching from `sass::sass_file()` to `xfun::read_utf8()` (in 7515b16f86d9f8180fac8c2dd7f149ef610bc874) should fix all the issues your seeing.

Reply:
Yes now all examples are working